### PR TITLE
Replace import equals declarations with const and type alias statements

### PR DESCRIPTION
**Pull request by hayes**
Currently the typedefs file uses a lot of import-equals statements to re-export aliased types and variables.  

Unfortunately, babel does not support import-equals syntax.  Import equals is part of typescript modules which are being replaced with es6 modules. 

This PR replaces importEquals statements in the thrift-server output with consts and type aliases.  I did not update the apache output because it does not have an existing clean way to separate from classes an enums.  Without that separation there is no way to determine if the statement needs to be a type alias or a const declaration.

### CODE REVIEW

File: src/tests/unit/fixtures/thrift-server/generated-strict/com/test/common/typedefs.ts [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-export import MoreOptionsArgs = __NAMESPACE__.OtherCommonUnionArgs;
-export import IMoreOptionsWithOption1Args = __NAMESPACE__.IOtherCommonUnionWithOption1Args;
-export import IMoreOptionsWithOption2Args = __NAMESPACE__.IOtherCommonUnionWithOption2Args;
-export import MoreOptionsCodec = __NAMESPACE__.OtherCommonUnionCodec;
+export type ICommonStruct = com_test_shared.ISharedStruct;
+export type ICommonStructArgs = com_test_shared.ISharedStructArgs;
+export const CommonStruct = com_test_shared.SharedStruct;
+export const CommonStructCodec = com_test_shared.SharedStructCodec;
+export type CommonUnionType = com_test_shared.SharedUnionType;
+export const CommonUnion = com_test_shared.SharedUnion;
```
Comment (hayes) [inline src/tests/unit/fixtures/thrift-server/generated-strict/com/test/common/typedefs.ts]: This should be a type alias.

File: src/tests/unit/fixtures/thrift-server/generated-strict/com/test/common/typedefs.ts [hunk 2]
```diff
@@ -1,32 +1,32 @@
 /* tslint:disable */
 /*
- * Autogenerated by @creditkarma/thrift-typescript v{{VERSION}}
+ * Autogenerated by @creditkarma/thrift-typescript v3.5.0
```
Comment (hayes) [inline src/tests/unit/fixtures/thrift-server/generated-strict/com/test/common/typedefs.ts]:
```suggestion
 * Autogenerated by @creditkarma/thrift-typescript v{{VERSION}}
```

**Comment by kevin-greene-ck** [comment]
@hayes Thanks for taking this on. One regression I'm getting is that for a `typedef` where the `definitionType` is an `enum` you reexport as a `const` which works fine when using the `enum` as a value, but you can also use an `enum` as a type which doesn't work now. I think you can reexport enums twice, once as a `const` and once as a `type` and that seems to work in my brief testing.

**Comment by kevin-greene-ck** [comment]
It would be easier to properly test this library if it was in the `thrift-server` mono repo. Maybe...

**Comment by hayes** [comment]
@kevin-greene-ck good catch! added some tests to cover that case, and updated enum export to export both type and const.  Looks like that works for using as both a type and value when imported!

**Comment by kevin-greene-ck** [comment]
@hayes Awesome, quick turn around. I'll spend a little time testing but the code looks great and I'll hopefully merge and release today.

**Comment by hayes** [comment]
thanks!

**Comment by hayes** [comment]
Going to push another change.  I discovered that the import order matters with the way that babel compiles the the const exports for some use cases.

**Comment by hayes** [comment]
@kevin-greene-ck I just pushed a small change to move the typedef imports to the bottom of the index.ts imports. 

This was because the way that babel compiles imports causes some exports from typedefs to be undefined.  This is because of the circular dependency created by every non-index file importing index.  index initially has an empty export object, and this is used to set the const values in the typedef  file. By importing typedefs last, we avoid these issues.

**Pull request merged**